node/src/main/scala/org/bitcoins/node/Node.scala

```diff
@@ -107,22 +142,10 @@ trait Node extends NodeApi with ChainQueryApi with P2PLogger {
     * object. Internally in [[org.bitcoins.node.networking.P2PClient p2p client]] you will see that
     * the [[ChainApi chain api]] is updated inside of the p2p client
     */
-  lazy val clients: Vector[P2PClient] = {
-    val peerMsgRecvs: Vector[PeerMessageReceiver] =
-      peers.map(x => PeerMessageReceiver.newReceiver(node = this, peer = x))
-    val zipped = peers.zip(peerMsgRecvs)
-    val p2p = zipped.map { case (peer, peerMsgRecv) =>
-      P2PClient(context = system,
-                peer = peer,
-                peerMessageReceiver = peerMsgRecv,
-                onReconnect = sync)
-    }
-    p2p
-  }
+  def clients: Vector[P2PClient] = peerData.values.map(_.client).toVector
```

Comment:
So this needed to be changed to a `def` as we can dynamically add or remove `P2PClient`s from the underlying `peerData` data structure right?

Reply:
Yes